packages/eslint-plugin/src/rules/sort-ngmodule-metadata-arrays.ts

```diff
@@ -16,42 +16,124 @@ export default createESLintRule<Options, MessageIds>({
         'Ensures ASC alphabetical order for `NgModule` metadata arrays for easy visual scanning',
       category: 'Best Practices',
       recommended: false,
+      suggestion: true,
     },
     fixable: 'code',
     schema: [],
     messages: {
       sortNgmoduleMetadataArrays:
         '`NgModule` metadata arrays should be sorted in ASC alphabetical order',
+      suggestFix: 'Sort members (WARNING! comments will probably be messed up)',
     },
   },
   defaultOptions: [],
   create(context) {
+    const sourceCode = context.getSourceCode();
     const metadataPropertyPattern =
       /^(bootstrap|declarations|entryComponents|exports|imports|providers|schemas)$/;
 
     return {
       [`${Selectors.MODULE_CLASS_DECORATOR} ${Selectors.metadataProperty(
         metadataPropertyPattern,
       )} ArrayExpression`]({ elements }: TSESTree.ArrayExpression) {
-        const unorderedNodes = elements
-          .filter(TSESLintASTUtils.isIdentifier)
-          .map((current, index, list) => [current, list[index + 1]])
-          .find(([current, next]) => {
-            return next && current.name.localeCompare(next.name) === 1;
-          });
+        const [expressions, identifiers] = partition(
+          elements,
+          TSESLintASTUtils.isIdentifier,
+        );
+        const firstUnorderedNode = getFirstUnderedNode(identifiers);
 
-        if (!unorderedNodes) return;
+        if (!firstUnorderedNode) {
+          return;
+        }
 
-        const [unorderedNode, nextNode] = unorderedNodes;
+        const fix: TSESLint.ReportFixFunction = (fixer) =>
+          getFixes(sourceCode, fixer, elements, expressions, identifiers);
         context.report({
-          node: unorderedNode,
+          node: firstUnorderedNode,
           messageId: 'sortNgmoduleMetadataArrays',
-          fix: (fixer) => [
-            fixer.replaceText(unorderedNode, nextNode.name),
-            fixer.replaceText(nextNode, unorderedNode.name),
-          ],
+          ...(hasCommentsWithinElements(sourceCode, elements)
+            ? { suggest: [{ messageId: 'suggestFix', fix }] }
+            : { fix }),
```

Comment:
I even tried to use `fix`, even if there were comments, but for god's sake, besides being very costly to maintain such an implementation, it's also very prone to failure.